src/test/java/ru/training/at/hw7/pages/MetalAndColorsPage.java

```diff
@@ -0,0 +1,71 @@
+package ru.training.at.hw7.pages;
+
+import com.epam.jdi.light.elements.common.UIElement;
+import com.epam.jdi.light.elements.complex.Checklist;
+import com.epam.jdi.light.elements.complex.dropdown.Dropdown;
+import com.epam.jdi.light.elements.composite.WebPage;
+import com.epam.jdi.light.elements.pageobjects.annotations.Title;
+import com.epam.jdi.light.elements.pageobjects.annotations.Url;
+import com.epam.jdi.light.elements.pageobjects.annotations.locators.JDropdown;
+import com.epam.jdi.light.elements.pageobjects.annotations.locators.UI;
+import com.epam.jdi.light.ui.html.elements.common.Button;
+import com.epam.jdi.light.ui.html.elements.complex.RadioButtons;
+import java.util.Arrays;
+import ru.training.at.hw7.dataProvider.ParametersForCheckingMetalAndColorsForm;
+
+@Url("/metals-colors.html")
+@Title ("Metals & Colors")
+public class MetalAndColorsPage extends WebPage {
+
+    @UI("[name=custom_radio_odd]")
+    public static RadioButtons oddSummaryRadioButtons;
+    @UI ("[name=custom_radio_even]")
+    public static RadioButtons evenSummaryRadioButtons;
+    @UI ("#elements-checklist input")
+    public static Checklist elementsCheckboxes;
+    @JDropdown(root = "div[ui=dropdown]",
+               value = ".filter-option",
+               list = "li",
+               expand = ".caret")
+    public static Dropdown colorsList;
+    @JDropdown (root = "div[ui=combobox]",
+                value = ".filter-option",
+                list = "li",
+                expand = ".caret")
+    public static Dropdown metalsList;
+    @UI ("#vegetables")
+    public static Button vegetablesDropdownButton;
+    @JDropdown (root = "div[ui=droplist]",
+                value = ".filter-option",
+                list = "li",
+                expand = ".caret")
+    public static Dropdown vegetablesList;
+    @UI ("#submit-button")
+    public static Button submitButton;
+    @UI (".panel-body-list.results")
+    public static UIElement resultSection;
+
+    public void fillForm(ParametersForCheckingMetalAndColorsForm parametersForCheckingMetalAndColorsForm) {
+        oddSummaryRadioButtons.select(String.valueOf(parametersForCheckingMetalAndColorsForm.getSummary()[0]));
+        evenSummaryRadioButtons.select(String.valueOf(parametersForCheckingMetalAndColorsForm.getSummary()[1]));
+        elementsCheckboxes.select(parametersForCheckingMetalAndColorsForm.getElements());
+        colorsList.select(parametersForCheckingMetalAndColorsForm.getColor());
+        metalsList.select(parametersForCheckingMetalAndColorsForm.getMetals());
+        vegetablesDropdownButton.click();
+        vegetablesList.select("Vegetables");
+        vegetablesList.select(parametersForCheckingMetalAndColorsForm.getVegetables());
+    }
+
+    public void submit() {
+        submitButton.click();
+    }
+
+    public void shouldContainProvidedDataInResultSection(
```

Comment:
1. At this point we can see that on this page there is a Form and there is a log section. Moreover Form have existing methods for fill and submit. So it's better to divide this Page parts.
2. Usually PageObject pattern is used to create a logic to describe page elements and user actions. For all checks that we want to perform during test execution should be located in Steps.

Reply:
Improved. Now page elements and actions are located in separate classes.